dev/io.openliberty.checkpoint_fat_mail/fat/src/io/openliberty/checkpoint/fat/IMAPTest.java

```diff
@@ -0,0 +1,219 @@
+
+    }
+
+    /**
+     * The BeforeClass method sets up the GreenMail test server, so that the "server" is waiting
+     * before even the Liberty class is set up.
+     *
+     * @throws UserException
+     * @throws Exception
+     */
+    @BeforeClass
+    public static void startGreenMail() throws Exception {
+        setupApp();
+        if (server.isStarted() != true) {
+            server.setCheckpoint(CheckpointPhase.AFTER_APP_START, false, null);
+            server.startServer();
+            // Update the mailSession config with correct password
+            updateVariableConfig(server, "imap_password", "imapPa$$word");
+
+            server.checkpointRestore();
+            // Pause for application to start properly and server to say it's listening on ports
+            server.waitForStringInLog("port " + server.getHttpDefaultPort());
+        }
+        // Get the current JVM options and add to it.
+        Map<String, String> jvmOptions = server.getJvmOptionsAsMap();
+
+        // Set the e-mail subject string that will be used in MimeMessage stored on the imapServer
+        String testSubjectString = "Sent from Liberty JavaMail";
+        // Set the e-mail body string for the same MimeMessage
+        String testBodyString = "Test mail sent by GreenMail";
+        // Create a ServerSetup with a port and host name and protocol type passed to it
+        // which is then used to create the actual GreenMail server.
+        int imapPort = Integer.getInteger("imap_port"); // As per server.xml
+        // Need to add -D to the start of the property name
+        System.out.println("Starting IMAP server on port " + imapPort);
+        ServerSetup imapSetup = new ServerSetup(imapPort, "localhost", "imap");
+        imapServer = new GreenMail(imapSetup);
+        // Start the imapServer, the GreenMail server is now listening for connections
+        imapServer.start();
+
+        // Add a user account to the mailTestServer by setting a user with email, userid, and password
+        // of the user. For this test purpose email and userid are the same
+        GreenMailUser setupTestUser = imapServer.setUser("imap@testserver.com",
+                                                         "imap@testserver.com", "imapPa$$word");
+        //Create the user on the mailTestServer
+        setupTestUser.create();
+        imapServer.setUser("imaps@testserver.com", "imapsPa$$word");
+
+        // After the server and user are setup, a MimeMessage is used to
+        // to store a message under the user's inbox folder on the GreenMail server
+        MimeMessage message = new MimeMessage((javax.mail.Session) null);
+        try {
+            message.setFrom(new InternetAddress("imaps@testserver.com"));
+            message.addRecipient(Message.RecipientType.TO, new InternetAddress("imaps@testserver.com"));
+            message.setSubject("Sent from Liberty JavaMail");
+            message.setText("Test mail sent by GreenMail");
```

Comment:
looks like these are the same strings as the unused ones above.  I suggest you remove the variables above.